### DISCUSSION TITLE
Shorter names and climate entry

### DISCUSSION THREAD

**Discussion by Linwood-F** [esphome-econet/esphome-econet#131] [Q&A] [answered]
This may more properly be a general esphome question but it is a bit buried in how the package is put together.

First thank you, and this desire should bear no negative reflection on the package.  Please do not take it as such. 

I like shorter names.  I have only one, and would like the sensors and entities and such to be simply "econet" or "hphw" or "hpwh" or some such, not the combinations.

So I set this in substitutions: 

    friendly_name: econet
    name: econet

This gets all the entities named like "econet fan speed" and "sensor.econet_fan_speed".  Great. 

Except one: while the name is "econet" the entity_id is "climate.econet_econet", with the name doubled. 

The definition shows "name: None" so I am not quite sure why it is doubling.  Anyone have an idea how to fix? 

On a slightly related note, and I am pretty sure the answer is no, is there any way to override specific sensor definitions to remove them, e.g. to mark "Internal: true" for example.  The "!extend" doesn't seem viable (maybe just that it requires an id which is not present), if I just put in a new definition after the package reference it is an additional sensor... yes, I can go in later and disable the sensor, but it would be nice if ESPHOME allowed you to override sensor characteristics pulled in from packages.  Is there a syntax I am missing?   If every sensor had an ID would !extend work?   And might that somehow allow me to override the name in climate? 

Again, thank you for this tool, it's the first package I have used for esphome, been learning a lot working with it.  Even ordered another set of M5Stack stuff for a different project, I had not run across their stuff before. 

Linwood

**Comment by tronikos** [comment, marked as answer]
For me the entity id is: `climate.econet_hpwh_water_heater`. It seems it combines name and friendly name. No idea how to change that. You should be able to rename it inside Home Assistant though.

Yes if every sensor had an ID, `!extend` would work. I'm adding ids in https://github.com/esphome-econet/esphome-econet/pull/132

**Comment by Linwood-F** [comment]
Ah, great.  Thank you.
